src/frontend/apps/impress/src/features/docs/doc-tree/components/DocTreeItemActions.tsx

```diff
@@ -143,48 +149,96 @@ export const DocTreeItemActions = ({
     }
   };
 
+  const {
+    handleMoreOptionsClick,
+    handleMoreOptionsKeyDown,
+    handleAddChildClick,
+    handleAddChildKeyDown,
+  } = useDocTreeItemHandlers({
+    isOpen,
+    onOpenChange,
+    createChildDoc,
+    docId: doc.id,
+  });
```

Comment:
It make it a bit complex to follow what is going on here.

I think it would be more clear with 2 dedicated components, and in these components the interactions that belong to them.

For this one, it could be `ButtonAddChildDoc`:
https://github.com/suitenumerique/docs/blob/154abdd451f8a92c2a21abc2a4727a3c70c9099d/src/frontend/apps/impress/src/features/docs/doc-tree/components/DocTreeItemActions.tsx#L223-L244


For this one, it could be `ButtonMoreOptions`:
https://github.com/suitenumerique/docs/blob/154abdd451f8a92c2a21abc2a4727a3c70c9099d/src/frontend/apps/impress/src/features/docs/doc-tree/components/DocTreeItemActions.tsx#L205-L220